vars/importTerraformModules.groovy

```diff
@@ -0,0 +1,210 @@
+#!groovy
+
+// Section to import Azure resources deployed using azurerm_template_deployment in to Terraform native resources
+// USAGE:
+
+// withPipeline(type, product, component) {	
+//   importTerraformModules()
+// }
+
+import groovy.json.JsonSlurper
+import uk.gov.hmcts.contino.azure.Az
+
+//can be run only inside withSubscription
+def call(String subscription, String environment, String product, tags) {
+    echo "Importing Terraform modules"
+
+    stageWithAgent("Import Terraform Modules", product) {
+        def jsonSlurper = new JsonSlurper()
+
+        def tfImport = "terraform import -var 'common_tags=${tags}' -var 'env=${environment}' -var 'product=${product}'" + 
+                        (fileExists("${environment}.tfvars") ? " -var-file=${environment}.tfvars" : "")
+
+        Closure az = { cmd -> return sh(script: "env AZURE_CONFIG_DIR=/opt/jenkins/.azure-$subscription az $cmd", returnStdout: true).trim() }
+
+        importModules = new ImportTerraformModules(this, environment, product, tags, az)
+        importModules.initialise(tfImport)
+
+        String stateJsonString =  sh(script: "terraform show -json", returnStdout: true).trim()
+        def stateJsonObj = jsonSlurper.parseText(stateJsonString)
+
+        // Create a backup/snapshot of the state file
+        tfstate = "${product}/${environment}/terraform.tfstate"
+        echo "Backup state file - ${tfstate}"
+        def snapShot = az "storage blob snapshot --container-name=${env.STORE_sa_container_name_template}${environment} --name=${tfstate} --account-name=${env.STORE_sa_name_template}${subscription}"
```

Comment:
do we not end up in snapshots on every run even after migrating ? have u tested that ?

Reply:
Updated logic, please review.